lib/mayaUsd/ufe/UsdPointInstanceModifierBase.h

```diff
@@ -51,36 +74,45 @@ template <class UfeValueType, class UsdValueType>
 class MAYAUSD_CORE_PUBLIC UsdPointInstanceModifierBase
 {
 public:
+    typedef std::shared_ptr<UsdPointInstanceBatch<UsdValueType>> Batch;
+    typedef std::unordered_map<Ufe::Path, Batch>                 Batches;
+
     UsdPointInstanceModifierBase()
         : _prim()
         , _instanceIndex(-1)
     {
     }
 
-    UsdPointInstanceModifierBase(const PXR_NS::UsdPrim prim, int instanceIndex)
-    {
-        setPrimAndInstanceIndex(prim, instanceIndex);
-    }
-
     virtual ~UsdPointInstanceModifierBase() = default;
 
-    bool setPrimAndInstanceIndex(const PXR_NS::UsdPrim prim, int instanceIndex)
+    bool setSceneItem(const UsdSceneItem::Ptr& sceneItem)
     {
         _prim = PXR_NS::UsdPrim();
         _instanceIndex = -1;
+        _path = Ufe::Path();
+
+        auto prim = sceneItem->prim();
+        auto instanceIndex = sceneItem->instanceIndex();
 
         if (!PXR_NS::UsdGeomPointInstancer(prim) || instanceIndex < 0) {
             return false;
         }
 
         _prim = prim;
         _instanceIndex = instanceIndex;
+        _path = sceneItem->path();
 
         return true;
     }
 
     PXR_NS::UsdPrim prim() const { return _prim; };
 
+    const Ufe::Path& path() const { return _path; };
+
+    Ufe::Path pointInstancerPath() const { return _path.pop(); };
+
+    virtual Batches& batches() = 0;
```

Comment:
Doesn't feel very kosher to give access to this to the outside world. I will probably learn later why it's needed, but for now, it feels like detail like this should be protected.

Reply:
Excellent point, done.